dbt_meshify/dbt_meshify.py

```diff
@@ -34,10 +47,127 @@ def add_model_contract_to_yml(self, model_name: str, model_catalog: CatalogTable
         # update the model entry in the full yml file
         # if no entries exist, add the model entry
         # otherwise, update the existing model entry in place
-        model_ordered_dict.update(model_yml)
-        # remove any keys with None values
-        model_ordered_dict = {k: v for k, v in model_ordered_dict.items() if v is not None}
-        models[model_name] = model_ordered_dict
+        processed = self.process_model_yml(model_yml)
+        models[model_name] = processed
+
+        full_yml_dict["models"] = list(models.values())
+        return full_yml_dict
+    
+    def add_model_version_to_yml(self, model_name, full_yml_dict, prerelease: Optional[bool] = False, defined_in: Optional[str] = None) -> Dict[str, Any]:
+        """Adds a model version to the model's yaml"""
+        # set up yml order
+
+        models = { model['name']: model for model in full_yml_dict['models'] } if full_yml_dict else {}
+        model_yml = models.get(model_name) or {"name": model_name, "latest_version": 0, "versions": []}
+        # add the version to the model yml entry
+        versions_list = model_yml.get("versions") or []
+        latest_version = model_yml.get("latest_version") or 0
+        version_dict = {}
+        if not versions_list:
+            version_dict["v"] = 1
+            latest_version +=1 
+        # if the model has versions, add the next version 
+        # if prerelease flag is true, do not increment the latest_version 
+        elif prerelease:
+            version_dict = {"v": latest_version + 1}
+        else:
+            version_dict = {"v": latest_version + 1}
+            latest_version += 1
+        # add the defined_in key if it exists
+        if defined_in:
+            version_dict["defined_in"] = defined_in
+        # add the version to the model yml entry
+        versions_list.append(version_dict)
+        # update the latest version in the model yml entry
+        model_yml["versions"] = versions_list
+        model_yml["latest_version"] = latest_version
+
+        processed = self.process_model_yml(model_yml)
+        models[model_name] = processed
```

Comment:
⚠️ Non-blocking feedback

I'm not quite sure where this ought to go. I know that there's a need for an ergonomic abstraction for working with YML and saving model properties. I suspect we'll be well served by letting this PR ride, and then coming back to redesign this abstraction.

Reply:
do you have early thoughts here? happy to take a crack at some of these refactoring tasks this week